subworkflows/local/minimap2_subworkflow.nf

```diff
@@ -19,8 +19,15 @@ workflow MINIMAP2_SUBWORKFLOW {
     ch_versions = ch_versions.mix(MINIMAP2_INDEX.out.versions)
     ch_index = MINIMAP2_INDEX.out.index
 
+    // match index to the corresponding reads
+    ch_reads = reads.map{meta, reads -> [meta.id, meta,reads]}
+    ch_mapping = ch_index
+        .map{meta, index -> [meta.id, meta, index]}
+        .combine(ch_reads, by:0)
```

Comment:
does this enforce that they have the same id?

Reply:
yes, so that the corresponding reads are mapping to the corresponding assembly. Otherwise random sets of reads would be mapped to random sets of assemblies, which I found when I added the mapping stats report where 0% of reads were mapping back to some assemblies because the wrong samples were queued together

Reply:
I'm not sure if the tests catch this and if there is a way to set that up other than expecting some threshold of reads mapped to the assembly...